crates/mipsy/src/main.rs

```diff
@@ -105,10 +105,20 @@ where
 
 fn get_input_int(name: &str) -> Option<i32> {
     loop {
-        let result: Result<i64, _> = try_read!();
+        let result: Result<i128, _> = try_read!();
 
         match result {
-            Ok(n) => return Some(n as i32),
+            Ok(n) => {
+                if n < std::i32::MIN as i128 || n > std::i32::MAX as i128 {
```

Comment:
[i32 impl `TryFrom<i128>`](https://doc.rust-lang.org/stable/std/primitive.i32.html#impl-TryFrom<i128>) will probably work better here

Reply:
lol idk how I forgot about that